### PR TITLE
Descriptors

### PR DESCRIPTION
**Pull request by davida26**
- all 8 descriptors connected see search page target to targert
- descriptor dynamic routes
- search checkboxes for each descriptor to view
- search fix to error on filterby
- optimizations to components

### CODE REVIEW

File: frontend/src/components/search/adverseEventToTarget.tsx [hunk 1]
```diff
@@ -38,19 +39,13 @@ function AdverseEventToTargetSearch() {
...
       }
 
-      const filterByCallback = (option, props) => {
-        const query = props.text.toLowerCase();
-        const adverseEventIdMatch = option.adverseEventId ? option.adverseEventId.toLowerCase().includes(query) : false;
-        const meddraIdMatch = option.meddraId ? option.meddraId.toLowerCase().includes(query) : false;
-        
-        return adverseEventIdMatch || meddraIdMatch;
-      };
+      const filterByFields = ['adverseEventId'];
```
Comment (nsheely) [inline frontend/src/components/search/adverseEventToTarget.tsx:42]: Should this list have 'adverseEventId' and 'meddraId'?

File: frontend/src/components/search/targetToAdverseEventSearch.tsx [hunk 2]
```diff
@@ -38,19 +37,14 @@ function TargetToAESearch() {
         router.push(`targetToAdverseEvents/${selectedTypeAhead[0].symbol}`)
       }
 
-      const filterByCallback = (option, props) => {
-        const query = props.text.toLowerCase();
-        const symbolMatch = option.symbol ? option.symbol.toLowerCase().includes(query) : false;
-        const descriptionMatch = option.description ? option.description.toLowerCase().includes(query) : false;
-        return symbolMatch || descriptionMatch;
-      };
+      const filterByFields = ['symbol'];
```
Comment (nsheely) [inline frontend/src/components/search/targetToAdverseEventSearch.tsx:40]: should this have symbol and description?

**Review by alyssabedard — APPROVED** [review]
niceee

File: frontend/src/components/search/targetToTargetSimilaritySearch.tsx [hunk 3]
```diff
@@ -31,28 +31,39 @@ function TargetToTargetSimilaritySearch() {
...
+      const openDescriptorData = () => {
+        checkedValues.forEach((value) => handleOpenNewTab(value))
+      }
+
+      const handleOpenNewTab = (value) => {
+        const url = `/similarity/${value}/${selectedTypeAhead[0].symbol}`;
+        window.open(url, "_blank");
       };
       
+      const filterByFields = ["symbol"];
```
Comment (nsheely) [inline frontend/src/components/search/targetToTargetSimilaritySearch.tsx:60]: Did we want to filter by description here too?

**Review by nsheely — CHANGES_REQUESTED** [review]

File: frontend/src/components/search/targetToAdverseEventSearch.tsx (hunk 2 above)
Comment (nsheely) [reply frontend/src/components/search/targetToAdverseEventSearch.tsx:40]: tested and the search still works

File: frontend/src/components/search/targetToTargetSimilaritySearch.tsx (hunk 3 above)
Comment (nsheely) [reply frontend/src/components/search/targetToTargetSimilaritySearch.tsx:60]: tested and the search still works

**Review by nsheely — APPROVED** [review]
tested and it seems like what I was concerned about works

**Pull request merged**